src/results_assembly/pfocr.js

```diff
@@ -0,0 +1,286 @@
+  const matchedTrapiResults = new Set();
+
+  const allGenesInAllFigures = figures.reduce((set, fig) => {
+    fig.associatedWith.mentions.genes.ncbigene.forEach((gene) => set.add(gene));
+    return set;
+  }, new Set());
+
+  for (const trapiResult of allTrapiResults) {
+    // No figures match this result
+    if (!figuresByCuries[trapiResultToCurieSet.get(trapiResult)]) continue;
+
+    const resultCuries = new Set();
+    const resultMatchableQNodeIDs = new Set();
+    [...matchableQNodeIDs].forEach((qNodeID) => {
+      trapiResult.node_bindings[qNodeID]
+        .map((node_binding) => node_binding.id)
+        .filter((curie) => curie.startsWith('NCBIGene:'))
+        .forEach((curie) => {
+          resultCuries.add(curie.replace('NCBIGene:', ''));
+          resultMatchableQNodeIDs.add(qNodeID);
+        });
+    });
+    if (resultMatchableQNodeIDs.size < 2) continue;
+
+    (figuresByCuries[trapiResultToCurieSet.get(trapiResult)] ?? []).forEach((figure, i) => {
+      if (!trapiResult.hasOwnProperty('pfocr')) {
+        trapiResult.pfocr = [];
+      }
+
+      const figureCurieSet = new Set(figure.associatedWith.mentions.genes.ncbigene);
+      const resultGenesInFigure = intersection(resultCuries, figureCurieSet);
+
+      const matchedQNodes = [...matchableQNodeIDs].filter((matchableQNodeID) => {
+        const currentQNodeCurieSet = new Set(
+          trapiResult.node_bindings[matchableQNodeID].map((node_binding) => node_binding.id),
+        );
+
+        return (
+          intersection(currentQNodeCurieSet, new Set([...resultGenesInFigure].map((geneID) => `NCBIGene:${geneID}`)))
+            .size > 0
+        );
+      });
+
+      // If we've matched on 2 curies, but we haven't actually matched on multiple nodes
+      if (matchedQNodes.length < 2) return;
+
+      const otherGenesInFigure = figureCurieSet.size - resultGenesInFigure.size;
+
+      let resultGenesInOtherFigures = [...resultCuries].filter((gene) => {
+        return figures.some((fig) => fig.associatedWith.mentions.genes.ncbigene.includes(gene));
+      }).length;
+      let otherGenesInOtherFigures = [...allGenesInAllFigures].filter((gene) => {
+        return !resultCuries.has(gene) && !figureCurieSet.has(gene);
+      }).size;
+
+      trapiResult.pfocr.push({
+        figureUrl: figure.associatedWith.figureUrl,
+        pmc: figure.associatedWith.pmc,
+        // TODO: do we want to include figure title? Note: this would need to be added to queryBody.
+        //title: figure.associatedWith.title,
+        nodes: matchedQNodes,
+        matchedCuries: [...resultGenesInFigure].map((geneID) => `NCBIGene:${geneID}`),
+        score:
+          1 -
+          parseFloat(
+            Analyze([
+              [resultGenesInFigure.size, resultGenesInOtherFigures],
+              [otherGenesInFigure, otherGenesInOtherFigures],
+            ]).pValue,
+          ),
+      });
+      matchedTrapiResults.add(trapiResult);
+    });
```

Comment:
@andrewsu This excerpt contains the scoring code.
- 207: resultGenesInFigure
- 223: otherGenesInFigure
- 225: resultGenesInOtherFigures
- 228: otherGenesInOtherFigures

Reply:
(note that I've since pushed a fix which replaces a `.size` with `.length` on line 230, which fixes that value previously being undefined)